<launch>

  <!-- Run wpb_mani core-->
  <node pkg="wpb_mani_bringup" type="wpb_mani_core" name="wpb_mani_core" output="screen">
    <param name="serial_port" type="string" value="/dev/ftdi"/> 
    <rosparam file="$(find wpb_mani_bringup)/config/wpb_mani.yaml" command="load" />
  </node>

  <!--- Run Rplidar -->
  <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
    <param name="serial_port"         type="string" value="/dev/rplidar"/>  
    <param name="serial_baudrate"     type="int"    value="115200"/>
    <param name="frame_id"            type="string" value="laser"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
    <remap from="scan" to="scan_raw"/>
  </node>

  <!-- Run lidar filter -->
  <node pkg="wpb_mani_bringup" type="wpb_mani_lidar_filter" name="wpb_mani_lidar_filter" output="screen">
    <param name="pub_topic" value="/scan"/>
  </node>

</launch>
